added Azure DevOps section (#15026)
aspnet/AspNetCore.Docs: aspnetcore/host-and-deploy/azure-apps/index.md

---
title: Deploy ASP.NET Core apps to Azure App Service

> author: bradygaster

Reviewer: btw -- This was only done to save U the pings! :smile: lol ... There's no author byline with my name on it. *Please do take the pings! ... They're alllllll yours!* 😀

Author: no worries or complaints from me @guardrex - I like being on there so they ping me. this is one of my areas so it makes sense that I get pinged when folks comment here.

Reviewer: Of course, by all means. I just wanted to make sure everyone was clear that I don't take credit for things that I don't write or work on. I was only here for the pings and to triage issues. This topic never had an author byline. Since this is your 👶, you might want to put a ...

```
By [Brady Gaster](https://twitter.com/bradygaster)
```

... under the H1.

Author: yeah I don't like adding those bylines. I figure use the author metadata.

See the [ASP.NET Core on App Service Dashboard](https://aspnetcoreon.azurewebsites.net/) for the version of ASP.NET Core available on Azure App service.

> Subscribe to the [App Service Announcements](https://github.com/Azure/app-service-announcements/) repository and watch the issues. The App Service team regularly posts updates in the issues section of the repository to announce updates and features arriving in App Service. 

Reviewer: Perhaps, say "monitor" instead of "watch"? ...

```suggestion
Subscribe to the [App Service Announcements](https://github.com/Azure/app-service-announcements/) repository and monitor the issues. The App Service team regularly posts announcements and scenarios arriving in App Service.
```

For more information, see <xref:security/data-protection/implementation/key-storage-providers>.

> <a name="deploy-aspnet-core-preview-release-to-azure-app-service"></a>

Reviewer: Just making sure that this is your plan: This section was always kept here to cover the **_next preview release_**. Next, it would go to cover preview of 3.1. If you're removing it tho permanently ... cool ... just calling out why it was always here, even when a release occurs.

## Deploy ASP.NET Core 3.0 to Azure App Service

> ASP.NET Core 3.0 is supported on Azure App Service. If you want to deploy preview releases of .NET Core 3.0+ in between major releases, or in the rare case during a deployment when the runtime is available but the SDK is not yet installed, use one of the techniques below to deploy an ASP.NET Core 3.0 (or later) app. 

Reviewer: ```suggestion
ASP.NET Core 3.0 is supported on Azure App Service. To deploy a preview release of a .NET Core version later than .NET Core 3.0, use one of the following techniques. These approaches are also used when the runtime is available but the SDK hasn't been installed on Azure App Service.
```

Alternatively, you could end the 2nd sentence with a colon and move the last sentence to a new paragraph (a one liner) immediately below the list as ...

> The preceding approaches are also used when the runtime is available but the SDK hasn't been installed on Azure App Service.

* [Specify the .NET Core SDK Version using Azure Pipelines](#specify-the-net-core-sdk-version-using-azure-pipelines)
* [Deploy a self-contained preview app](#deploy-a-self-contained-preview-app).
* [Use Docker with Web Apps for containers](#use-docker-with-web-apps-for-containers).
* [Install the preview site extension](#install-the-preview-site-extension).

### Specify the .NET Core SDK Version using Azure Pipelines

> Using [Azure App Service's CI/CD features](https://docs.microsoft.com/en-us/azure/app-service/deploy-continuous-deployment) makes it easy to set up a continuous integration build with Azure DevOps. Once this build has been created, a few additional tweaks can be made to not only "pin" your SDK version so that you're always confident to build against a consistent .NET Core SDK version. 

Reviewer: ```suggestion
Use [Azure App Service CI/CD scenarios](/azure/app-service/deploy-continuous-deployment) to set up a continuous integration build with Azure DevOps. After the Azure DevOps build is created, optionally configure the build to use a specific SDK version. 
```

btw - Management asked us not to speak of "features" but to try and say "scenarios." It doesn't always work out to be the best choice, so we still use "features" in a few spots.

> #### Specifying the .NET Core SDK Version

Reviewer: This is an unpaired fourth-level heading. Can you remove it?

If not, at least drop the gerund and go with sentence casing ...

```suggestion
#### Specify the .NET Core SDK version
```

> If you use the App Service deployment center to create your Azure DevOps build, the resulting build pipeline will have steps for `Restore`, `Build`, `Test`, and `Publish`. To specify the SDK version, click the "Add (+)" button in the Agent job list to add a new step. Search for **.NET Core SDK** in the search bar. 

Reviewer: ```suggestion
When using the App Service deployment center to create an Azure DevOps build, the resulting build pipeline has steps for `Restore`, `Build`, `Test`, and `Publish`. To specify the SDK version, select the **Add (+)** button in the Agent job list to add a new step. Search for **.NET Core SDK** in the search bar. 
```

Should "deployment center" be capitalized? ... it sounds like a proper noun.

![Add the .NET Core SDK step](index/add-sdk-step.png)

> Move the step into the first position in the build, so that the steps following it are sure to use that version of the .NET Core SDK. Specify the version of the .NET Core SDK you wish to use. In this example, the SDK has been set to the `3.0.100`. 

Reviewer: ```suggestion
Move the step into the first position in the build so that the steps following it use the specified version of the .NET Core SDK. Specify the version of the .NET Core SDK. In this example, the SDK is set to `3.0.100`.
```

![Completed SDK step](index/sdk-step-first-place.png)

> If you want to publish a self-contained app, you can specify in the `Publish` step to self-contain the app and the [RID](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog) you want to use. 

Reviewer: ```suggestion
To publish a [self-contained deployment (SCD)](/dotnet/core/deploying/#self-contained-deployments-scd), configure SCD in the `Publish` step and provide the [Runtime Identifier (RID)](/dotnet/core/rid-catalog).
```
